packages/test/test-end-to-end-tests/src/test/gcDataStoreRequests.spec.ts

```diff
@@ -0,0 +1,203 @@
+    createAndAttachContainer,
+    createLocalLoader,
+    OpProcessingController,
+} from "@fluidframework/test-utils";
+
+class TestDataObject extends DataObject {
+    public get _root() {
+        return this.root;
+    }
+
+    public get _runtime() {
+        return this.runtime;
+    }
+
+    public get _context() {
+        return this.context;
+    }
+}
+
+describe("GC Data Store Requests", () => {
+    const documentId = "garbageCollection";
+    const documentLoadUrl = `fluid-test://localhost/${documentId}`;
+    const codeDetails: IFluidCodeDetails = {
+        package: "garbageCollectionTestPackage",
+        config: {},
+    };
+    const factory = new DataObjectFactory(
+        "TestDataObject",
+        TestDataObject,
+        [],
+        []);
+    const runtimeOptions = {
+        generateSummaries: true,
+        enableWorker: false,
+        initialSummarizerDelayMs: 10,
+    };
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        factory,
+        [
+            [factory.type, Promise.resolve(factory)],
+        ],
+        undefined,
+        undefined,
+        runtimeOptions,
+    );
+
+    const IdleDetectionTime = 100;
+
+    const summaryConfig: ISummaryConfiguration = {
+        idleTime: IdleDetectionTime,
+
+        maxTime: IdleDetectionTime * 12,
+
+        // Snapshot if 1000 ops received since last snapshot.
+        maxOps: 1000,
+
+        // Wait 10 minutes for summary ack
+        maxAckWaitTime: 600000,
+    };
+    const serviceConfig: Partial<IClientConfiguration> = {
+        summary: summaryConfig,
+    };
+
+    let deltaConnectionServer: ILocalDeltaConnectionServer;
+    let urlResolver: LocalResolver;
+    let opProcessingController: OpProcessingController;
+    let container1: IContainer;
+
+    async function createContainer(): Promise<IContainer> {
+        const loader = createLocalLoader([[codeDetails, runtimeFactory]], deltaConnectionServer, urlResolver);
+        return createAndAttachContainer(codeDetails, loader, urlResolver.createCreateNewRequest(documentId));
+    }
+
+    async function loadContainer(): Promise<IContainer> {
+        const loader = createLocalLoader([[codeDetails, runtimeFactory]], deltaConnectionServer, urlResolver);
+        return loader.resolve({ url: documentLoadUrl });
+    }
+
+    async function waitForSummary(container: IContainer): Promise<string | undefined> {
+        let handle: string | undefined;
+        // wait for summary ack/nack
+        await new Promise((resolve, reject) => container.on("op", (op) => {
+            if (op.type === "summaryAck") {
+                handle = op.contents.handle;
+                resolve(true);
+            } else if (op.type === "summaryNack") {
+                reject(new Error("summaryNack"));
+            }
+        }));
+        return handle;
+    }
+
+    beforeEach(async () => {
+        deltaConnectionServer = LocalDeltaConnectionServer.create(undefined, serviceConfig);
```

Comment:
these tests should be server agnostic and use getTestDriver()

Reply:
The reason I didn't make it server agnostic is because it waits for summary op which takes approx. 5 seconds. So, the test would take like 20 seconds to complete for other servers. Local server's summary time can be tweaked by passing in service configs.